addons/mail/static/src/models/thread_view.js

```diff
@@ -61,6 +61,22 @@ registerModel({
             }
             this.threadCache.update({ hasLoadingFailed: false });
         },
+        /**
+         * Handles OWL update on the thread_view component.
+         */
+         onComponentUpdate() {
+            if (
+                this.thread &&
+                this.thread === this.messaging.inbox &&
+                this.lastThreadCounter > 0 && this.thread.counter === 0
+            ) {
+                this.env.services.effect.add({
+                    message: this.env._t("Congratulations, your inbox is empty!"),
+                    type: 'rainbow_man',
+                });
+            }
+            this.update({ lastThreadCounter: this.thread && this.thread.counter });
```

Comment:
Now that I see it, this part doesn't even depend on thread view either, just on thread. So maybe it could be moved to thread instead, as onchange instead of component update? (maybe conditionally that there is at least one thread view existing for the thread).